### PR TITLE
MAINT: Fix barrage of DataConversionWarnings (Fixes #19, Fixes #20)

### LINKED ISSUE #19: DataConversionWarning: column-vector passed when 1d array expected
I'm having difficulty figuring this error out. Here's the setup:

* I have a script that uses `keras.datasets.mnist`. It reshapes into a 2D array and a 1D vector of labels:

``` python
import tensorflow as tf
from tensorflow.keras.layers import Dense, Activation, Dropout
from tensorflow.keras.models import Sequential
from tensorflow.keras.datasets import mnist
from scikeras.wrappers import KerasClassifier

def get_keras():
    (X_train, y_train), _ = mnist.load_data()
    X_train, y_train = X_train[:100], y_train[:100]
    X_train = X_train.reshape(X_train.shape[0], 784).astype("float32") / 255
    return X_train, y_train

def _keras_build_fn(opt="sgd"):
    model = Sequential([Dense(512, input_shape=(784,)), Activation("relu"),
                        Dense(10), Activation("softmax")])
    model.compile(loss="binary_crossentropy", optimizer=opt, metrics=["accuracy"])
    return model

def test_keras():
    X, y = get_keras()
    assert X.ndim == 2 and X.shape[-1] == 784 and y.ndim == 1
    model = KerasClassifier(build_fn=_keras_build_fn)
    model.fit(X, y)

if __name__ == "__main__":
    test_keras()
```

* I run this script with `python _scikeras_bug.py`. Everything works.
* I run this script with `pytest _scikeras_bug.py`, and I get this error:

> sklearn.exceptions.DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().

Here's the traceback:

```
_scikeras_bug.py:36:
    model.fit(X, y)
../scikeras/scikeras/wrappers.py:503: in fit
    y, extra_args = self._pre_process_y(y)
../scikeras/scikeras/wrappers.py:742: in _pre_process_y
    y = encoder.fit_transform(y)
../../../anaconda3/envs/dask-ml-docs/lib/python3.6/site-packages/sklearn/preprocessing/_label.py:255: in fit_transform
    y = column_or_1d(y, warn=True)
../../../anaconda3/envs/dask-ml-docs/lib/python3.6/site-packages/sklearn/utils/validation.py:73: in inner_f
    return f(**kwargs)../../../anaconda3/envs/dask-ml-docs/lib/python3.6/site-packages/sklearn/utils/validation.py:843: DataConversionWarning
```

I've tried this out with two environments, and got the same error both times:

* pytest 5.4.3, scikeras 0.1.7, Tensorflow 2.2.0, Python 3.6.10
* pytest 6.0.0rc1, scikeras 0.1.7, Tensorflow 2.2.0, Python 3.8.3.

### LINKED ISSUE #20: MAINT: minimize warnings
- Review all warnings emerging from tests and try to minimize them.
- Block "known" warnings (i.e. ConversionWarning or warnings from tests that are supposed to fail).
- Avoid reshaping y when unneeded or add reshaping when the Keras output does not match the input to minimize DataConversionWarning s.

### PR DESCRIPTION
**Pull request by adriangb**
Fixes #19, Fixes #20

### CODE REVIEW

**Comment by adriangb** [comment]
@stsievert want to take a look at this and see if it allows you to remove the warnings capture in dask/dask-ml/issues/696?
